src/components/NewDatePicker/CalendarPicker/YearPickerModal.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '../../withCurrentUserPersonalDetails';
+import HeaderWithBackButton from '../../HeaderWithBackButton';
+import withLocalize, {withLocalizePropTypes} from '../../withLocalize';
+import CONST from '../../../CONST';
+import SelectionListRadio from '../../SelectionListRadio';
+import Modal from '../../Modal';
+import {radioListItemPropTypes} from '../../SelectionListRadio/selectionListRadioPropTypes';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+    ...withCurrentUserPersonalDetailsPropTypes,
+
+    /** Whether the modal is visible */
+    isVisible: PropTypes.bool.isRequired,
+
+    /** The list of years to render */
+    years: PropTypes.arrayOf(PropTypes.shape(radioListItemPropTypes)).isRequired,
+
+    /** Currently selected year */
+    currentYear: PropTypes.number.isRequired,
+
+    /** Function to call when the user selects a year */
+    onYearChange: PropTypes.func.isRequired,
+
+    /** Value for the text input */
+    textInputValue: PropTypes.string.isRequired,
+
+    /** Callback to fire when the text input changes */
+    onChangeText: PropTypes.func.isRequired,
+
+    /** Function to call when the user closes the year picker */
+    onClose: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    ...withCurrentUserPersonalDetailsDefaultProps,
+    isVisible: false,
+    years: [],
+    currentYear: new Date().getFullYear(),
+    onYearChange: () => {},
+    textInputValue: '',
+    onChangeText: () => {},
+    onClose: () => {},
+};
+
+function YearPickerModal(props) {
+    const headerMessage = props.textInputValue.trim() && !props.years.length ? props.translate('common.noResultsFound') : '';
```

Comment:
Cool, I like this approach. But that does not reset the search alone, because the modal is never unmounted, it just becomes invisible. So I also added an effect to clear the search text when it closes